src/core/middleware.js

```diff
@@ -45,7 +50,19 @@ export function prefixMiddleWare(req, res, next, { _config = config } = {}) {
     // We skip prepending an app if we'd previously prepended a lang and the
     // 2nd part of the URL is now a valid app.
     log.info('Application in URL is valid following prepending a lang.');
-  } else if (!hasValidClientApp) {
+  } else if (
+    (hasValidUrlException || isValidUrlException(URLParts[1], { _config }))
```

Comment:
If I delete `hasValidUrlException` from this statement, the tests still pass. Is it needed? If so, please add test coverage.